node/worker/worker.go

```diff
@@ -91,15 +85,26 @@ func (w *Worker) CommitTransactions(
 			txs.Pop()
 			continue
 		}
-		// Start executing the transaction
-		w.current.state.Prepare(tx.Hash(), common.Hash{}, len(w.current.txs))
 
 		if tx.ShardID() != w.chain.ShardID() {
 			txs.Shift()
 			continue
 		}
 
-		_, err := w.commitTransaction(tx, coinbase)
+		var err error
+
+		// Start executing the transaction
+		switch tx.(type) {
+		case *types.EthTransaction:
+			w.current.state.Prepare(tx.Hash(), common.Hash{}, len(w.current.ethTxs))
+			_, err = w.commitEthTransaction(tx.(*types.EthTransaction), coinbase)
+		case *types.Transaction:
+			w.current.state.Prepare(tx.Hash(), common.Hash{}, len(w.current.ethTxs)+len(w.current.txs))
```

Comment:
Is the transaction strictly sorted that Eth transaction goes before hmy transaction? Why do we want EthTransaction having higher priority than hmy transaction?

Reply:
Yea, we will process eth txn strictly before hmy txn. There no good reason for this, but we expect eth txn eventually overtake hmy txn as eth ecosystem and tools are more popular.